pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -138,3 +142,35 @@ func (hr *horizontalController) computeScaleAction(
 
 	return targetDesiredReplicas
 }
+
+func isScalingAllowed(autoscalerInternal *model.PodAutoscalerInternal, source datadoghq.DatadogPodAutoscalerValueSource, direction scaleDirection) (bool, string) {
```

Comment:
Is it used for the admission controller / vertical controller ?

Reply:
No this is horizontal, there's a dedicated one for vertical.